src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -126,9 +134,15 @@ public async void BeginInvokeDotNetFromJS(string callId, string assemblyName, st
             }
         }
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
```

Comment:
Weird to use DisposeAsync for this? Should we have an explicit event for disconnects?

Reply:
This seems cool.